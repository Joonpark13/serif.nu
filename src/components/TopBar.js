import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/styles';

const topBarTextColor = 'white';

export const styles = {
  title: {
    color: topBarTextColor,
  },
  icon: {
    color: topBarTextColor,
  },
  toolbar: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  topBarLeftSection: {
    display: 'flex',
    alignItems: 'center',
  },
};

function TopBar({ classes, menuAction }) {
  return (
    <AppBar position="static">
      <Toolbar className={classes.toolbar}>
        <div className={classes.topBarLeftSection}>
          <IconButton aria-label="Navigation" onClick={menuAction}>
            <MenuIcon className={classes.icon} />
          </IconButton>
          <Typography className={classes.title} variant="h5">Serif.nu</Typography>
        </div>

        <Typography className={classes.title}>Term: Fall 2019</Typography>
      </Toolbar>
    </AppBar>
  );
}

TopBar.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  menuAction: PropTypes.func.isRequired,
};

export { TopBar as UnstyledTopBar };

export default withStyles(styles)(TopBar);
