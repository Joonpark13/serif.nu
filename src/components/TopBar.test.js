import React from 'react';
import { shallow } from 'enzyme';
import TopBar from './TopBar';

describe('TopBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TopBar menuAction={() => {}} />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
