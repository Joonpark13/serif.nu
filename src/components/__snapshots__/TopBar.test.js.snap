// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<ForwardRef(WithStyles)
  position="static"
>
  <ForwardRef(WithStyles)
    className="makeStyles-toolbar-3"
  >
    <div
      className="makeStyles-topBarLeftSection-4"
    >
      <ForwardRef(WithStyles)
        aria-label="Navigation"
        onClick={[Function]}
      >
        <pure(Menu)
          className="makeStyles-icon-2"
        />
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-title-1"
        variant="h5"
      >
        Serif.nu
      </ForwardRef(WithStyles)>
    </div>
    <ForwardRef(WithStyles)
      className="makeStyles-title-1"
    >
      Term: Fall 2019
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
